lib/puppet/provider/oneview_volume_template/c7000.rb

```diff
@@ -24,7 +24,48 @@
 
   mk_resource_methods
 
+  def set_storage_pool
+    storage_pool_class = OneviewSDK.resource_named('StoragePool', @client.api_version)
+    uri = storage_pool_class.get_all(@client).first['uri']
+    @data['properties']['storagePool']['default'] = uri
+    @data['properties']['snapshotPool']['default'] = uri
+  end
+
+  def set_initial_scope
+    list = []
+    scope = OneviewSDK.resource_named('Scope', api_version)
+    list.push(scope.get_all(@client).first['uri'])
+    @data['initialScopeUris'] = list
+  end
+
+  def set_template_uri
+    @data = resource['data']
+    volume_template_class = OneviewSDK.resource_named('VolumeTemplate', @client.api_version)
+    resource['data']['rootTemplateUri'] = volume_template_class.get_all(@client, isRoot: true).first['uri']
```

Comment:
for suppose if there are 3 volume templates, then your new code will pick only 1st template among them. What if user want to pass the name of template and wants the volume to use that.

Reply:
Since we have only one root Template uri, the above code covers all scenarios.